File: SIH26011/backend/app/api/lidar.py
from fastapi import APIRouter, HTTPException
from typing import Dict, Any
from app.repositories.data_repository import repository
from app.schemas.pydantic_schemas import LiDARAnalysisResponse
from app.services.lidar_engine import lidar_engine

router = APIRouter(prefix="/api/lidar", tags=["LiDAR & Point Cloud"])

@router.post("/analyze", response_model=LiDARAnalysisResponse)
def analyze_lidar(data: Dict[str, Any]):
    prop_id = data.get("property_id")
    if not prop_id:
        raise HTTPException(status_code=400, detail="property_id is required.")
    
    analysis = lidar_engine.analyze_point_cloud(prop_id)
    return LiDARAnalysisResponse(**analysis.__dict__)

@router.get("/{property_id}/lidar-analysis", response_model=LiDARAnalysisResponse)
def get_lidar_analysis(property_id: str):
    analysis = lidar_engine.analyze_point_cloud(property_id)
    return LiDARAnalysisResponse(**analysis.__dict__)

File: backend/src/routes/lidar.js
import { Router } from "express";
import { lidarEngine } from "../services/lidarEngine.js";

const router = Router();

// Run LiDAR analysis
router.post("/analyze", (req, res) => {
  const propertyId = req.body?.property_id;

  if (!propertyId) {
    return res.status(400).json({
      detail: "property_id is required.",
    });
  }

  const pointCount =
    req.body?.point_count ?? 150000;

  const result =
    lidarEngine.analyzePointCloud(
      propertyId,
      pointCount
    );

  return res.json(result);
});

// Get LiDAR analysis for a property
router.get(
  "/:property_id/lidar-analysis",
  (req, res) => {
    const result =
      lidarEngine.analyzePointCloud(
        req.params.property_id
      );

    return res.json(result);
  }
);

export default router;